learn/advanced/filtering_and_faceted_search.md

```diff
@@ -102,10 +102,14 @@ The [`GET` route of the search endpoint](/reference/api/search.md#search-in-an-i
 
 String expressions combine conditions using the following filter operators and parentheses:
 
-- `NOT` only returns documents that do not satisfy a condition : `NOT genres = horror`
+- `NOT` returns all documents that do not satisfy a condition, including results missing the specified attribute. The expression `NOT genres = horror` returns all documents whose `genres` do not contain `horror` and all documents missing a `genres` field
 - `AND` operates by connecting two conditions and only returns documents that satisfy both of them: `genres = horror AND director = 'Jordan Peele'`
 - `OR` connects two conditions and returns results that satisfy at least one of them: `genres = horror OR genres = comedy`
 - `TO` is equivalent to `>= AND <=`. The expression `release_date 795484800 TO 972129600` translates to `release_date >= 795484800 AND release_date <= 972129600`
+- `IN [valueA, valueB]` selects all documents whose chosen field contains at least one of the specified values. The expression `genres IN [horror, comedy]` returns all documents whose `genres` includes either `horror`, `comedy`, or both
```

Comment:
Nice and concise :) I would try to make it explicit that IN[..] can contain 0 or more values inside the brackets, and not strictly 2. 

If 0 values are given to it, it selects 0 documents. But I don't know whether that information should be in the documentation, it might unnecessarily confuse some people I guess.

The syntax `value NOT IN [..]` is also not explained. That is fine by me since they can use `NOT value IN [..]` instead, which is equivalent. But I just wanted to remind you of it in case omitting it was not a conscious decision :-)

Reply:
Ah, good catch. I'll add a hint indicating `IN` accepts more than two values.

As for your other two comments, indeed, it was conscious decision. We tend to prioritise information that is useful in the majority of cases over exhaustiveness—being exhaustive for the sake of completeness leads to people not reading the full thing because "ain't nobody got time for that." 

Not documenting both `NOT value IN` and `value NOT IN` additionally follows my personal preference of avoiding giving users options that I cannot also present clear reasons to choose one over the other.